frontend/src/components/mine/ContactInfo/ViewPartyRelationships.js

```diff
@@ -139,21 +137,68 @@ export class ViewPartyRelationships extends Component {
 
     payload.start_date = values.start_date;
     payload.end_date = values.end_date;
-    payload.mine_tailings_storage_facility_guid = values.mine_tailings_storage_facility_guid;
+    payload.related_guid = values.related_guid;
 
     this.props.updatePartyRelationship(payload).then(() => {
-      this.props.fetchPartyRelationships(this.props.mine.guid);
+      this.props.fetchPartyRelationshipsByMineId(this.props.mine.guid);
       this.props.closeModal();
     });
   };
 
   removePartyRelationship = (event, mine_party_appt_guid) => {
     event.preventDefault();
     this.props.removePartyRelationship(mine_party_appt_guid).then(() => {
-      this.props.fetchPartyRelationships(this.props.mine.guid);
+      this.props.fetchPartyRelationshipsByMineId(this.props.mine.guid);
     });
   };
 
+  renderInactiveRelationships = (partyRelationships) => {
+    const activeRelationships = partyRelationships.filter(
+      (x) =>
+        x.end_date === "None" ||
+        (Date.parse(x.end_date) >= new Date() &&
+          (x.start_date === "None" || Date.parse(x.start_date) <= new Date()))
+    );
+    const inactiveRelationships = partyRelationships.filter(
+      (x) =>
+        x.end_date !== "None" &&
+        (Date.parse(x.end_date) <= new Date() ||
+          (x.start_date !== "None" && Date.parse(x.start_date) >= new Date()))
+    );
+
+    const activePartyRelationshipTypes = [
+      ...new Set(activeRelationships.map((x) => x.mine_party_appt_type_code)),
+    ];
+    const inactivePartyRelationshipTypes = [
+      ...new Set(inactiveRelationships.map((x) => x.mine_party_appt_type_code)),
+    ];
+
+    return inactivePartyRelationshipTypes
+      .filter((x) => !activePartyRelationshipTypes.includes(x))
```

Comment:
Is there overlap between `activePartyRelationshipTypes` and `inactivePartyRelationshipTypes`? It doesn't seem like there should be, so this filter should be unnecessary.

If that's the case, then `activePartyRelationshipTypes`, and by extension, `activeRelationships` shouldn't be needed.

Reply:
There can be overlap, yes. The intent is to render a single stub, for what could be multiple inactive contacts, but only if there is currently no active contact of that type.